### PR TITLE
VMS: fix library compatibility settings in util/mkdef.pl

### PR DESCRIPTION
**Pull request by levitte**
The regexp to parse the incoming version number was flawed, and since
we allow ourselves to add missing APIs in PATCH releases, the
compatibility settings still need to include the PATCH part of the
version number.

### CODE REVIEW

**Comment by levitte** [comment]
VMS build currently fail because of the flaw.

File: util/mkdef.pl [hunk 1]
```diff
@@ -386,7 +386,10 @@ sub writer_VMS {
 _____
 
     if (defined $version) {
-        my ($libvmajor, $libvminor) = $version =~ /^(\d+)_(\d+)$/;
+        print STDERR "DEBUG: \$version = $version\n";
```
Comment (mspncp) [inline util/mkdef.pl]: Is this a leftover or intended to go in?

Comment (levitte) [reply util/mkdef.pl]: Oops.
Leftover. Will fix

File: util/mkdef.pl [hunk 2]
```diff
@@ -386,7 +386,9 @@ sub writer_VMS {
 _____
 
     if (defined $version) {
-        my ($libvmajor, $libvminor) = $version =~ /^(\d+)_(\d+)$/;
+        $version =~ /^(\d+)\.(\d+)\.(\d+)/;
+        my $libvmajor = $1;
+        my $libvminor = $2 * 100 + $3;
```
Comment (mspncp) [inline util/mkdef.pl]: I didn't try to understand _why_ the minor and patch version need to be coded into `$libvminor` in this way. But the perl code looks good to me. So I'm just assuming that you know what you are doing... ;-)

**Review by mspncp — APPROVED** [review]
LGTM

File: util/mkdef.pl (hunk 2 above)
Comment (levitte) [reply util/mkdef.pl]: I do :-)

**Comment by levitte** [comment]
Merged.

d1c87578a29b9b6506ff8c11bdf3e5d63f784965 VMS: fix library compatibility settings in util/mkdef.pl

**Pull request closed without merging**